average.go

```diff
@@ -0,0 +1,18 @@
+package main
+
+import "fmt"
+
+func average(x []int) float64 {
+	var sum, N int
+	sum = 0
+	N = len(x)
+	for i := 0; i < N; i++ {
+		sum += x[i]
+	}
+	return float64(sum) / float64(N)
```

Comment:
What happens if N == 0 ?
Please add a check that slice length is greater than 0.